GVFS/GVFS.Common/Maintenance/PackfileMaintenanceStep.cs

```diff
@@ -0,0 +1,122 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class PackfileMaintenanceStep : GitMaintenanceStep

```

Comment:
You have an exellent explaination at the top of the PR.  Could we add that in comments here

> This is done in two steps:
> 
> git multi-pack-index expire: This deletes the pack-files whose objects appear in newer pack-files. The multi-pack-index prevents git from looking at these packs. Rewrites the multi-pack-index to no longer refer to these (deleted) packs.
> 
> git multi-pack-index repack --batch-size=<size> inspects packs covered by the multi-pack-index in modified-time order (ascending). Greedily selects a batch of packs whose file sizes are all less than "size", but that sum up to at least "size". Then generate a new pack-file containing the objects that are uniquely referenced by the multi-pack-index.